jscomp/others/js_dict.ml

```diff
@@ -56,3 +56,55 @@ let unsafeDeleteKey : string t -> string -> unit [@bs] = [%raw{|
      return 0
    }
 |}]
+
+(* external entries : 'a t -> (key * 'a) array = "Object.entries" [@@bs.val] (* ES2017 *) *)
+let entries dict =
+  let keys = keys dict in
+  let l = Js.Array.length keys in
+  (* TODO: Get rid of Obj.magic *)
+  let values = Obj.magic (Array.make l 0) in
```

Comment:
can you use `external unsafeCreate : int -> 'a array = "Array" [@@bs.new]` instead, we don't want to depend on `Array` here, not friendly for DCE